diff --git a/include/env_default.h b/include/env_default.h
index 014873e3..127d3606 100644
--- a/include/env_default.h
+++ b/include/env_default.h
@@ -127,13 +127,13 @@ const char default_environment[] = {
 #ifdef CONFIG_MTDPARTS_DEFAULT
 	"mtdparts="	CONFIG_MTDPARTS_DEFAULT		"\0"
 #endif
+#ifdef	CFG_EXTRA_ENV_SETTINGS
+	CFG_EXTRA_ENV_SETTINGS
+#endif
 #ifdef CONFIG_EXTRA_ENV_TEXT
 	/* This is created in the Makefile */
 	CONFIG_EXTRA_ENV_TEXT
 #endif
-#ifdef	CFG_EXTRA_ENV_SETTINGS
-	CFG_EXTRA_ENV_SETTINGS
-#endif
 #ifdef CONFIG_OF_SERIAL_BAUD
 	/* Padding for baudrate at the end when environment is writable */
 	"\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"
