diff --git a/arch/arm64/boot/dts/amlogic/Makefile b/arch/arm64/boot/dts/amlogic/Makefile
index 619dce79b020..14efe8bf34f9 100644
--- a/arch/arm64/boot/dts/amlogic/Makefile
+++ b/arch/arm64/boot/dts/amlogic/Makefile
@@ -18,6 +18,7 @@ dtb-$(CONFIG_ARCH_MESON) += meson-g12a-fbx8am-brcm.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12a-fbx8am-realtek.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12a-radxa-zero.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12a-sei510.dtb
+dtb-$(CONFIG_ARCH_MESON) += meson-g12a-superbird.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12a-u200.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12a-x96-max.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-g12b-a311d-bananapi-m2s.dtb
diff --git a/drivers/Kconfig b/drivers/Kconfig
index 4915a63866b0..c46a5912ca14 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -1,6 +1,8 @@
 # SPDX-License-Identifier: GPL-2.0
 menu "Device Drivers"
 
+source "drivers/amlogic/Kconfig"
+
 # Keep I/O buses first
 
 source "drivers/amba/Kconfig"
diff --git a/drivers/Makefile b/drivers/Makefile
index 8e1ffa4358d5..c43441e7024b 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -197,3 +197,5 @@ obj-$(CONFIG_DPLL)		+= dpll/
 
 obj-$(CONFIG_DIBS)		+= dibs/
 obj-$(CONFIG_S390)		+= s390/
+
+obj-$(CONFIG_AMLOGIC_DRIVER) 	+= amlogic/
diff --git a/drivers/amlogic/Kconfig b/drivers/amlogic/Kconfig
new file mode 100644
index 000000000000..81858945bd07
--- /dev/null
+++ b/drivers/amlogic/Kconfig
@@ -0,0 +1,30 @@
+#
+# Amlogic driver configuration
+#
+menuconfig AMLOGIC_DRIVER
+	bool "Amlogic Device Drivers"
+	default n
+	help
+		this option is provided for control amlogic
+		drivers, if you want to use amlogic driver
+		please open it
+
+if AMLOGIC_DRIVER
+
+config AMLOGIC_TOUCHSCREEN_TLSC6X
+	bool "TLSC6X Touchscreen"
+	depends on I2C
+	default n
+	help
+		Say Y here if you have tlsc6x touch panel.
+		If unsure, say N.
+
+config AMLOGIC_GPIO_IRQ
+	bool "Meson GPIO Interrupt Multiplexer Double Edge Support"
+	depends on IRQ_DOMAIN
+	depends on IRQ_DOMAIN_HIERARCHY
+	default n
+	help
+	  Support Meson SoC Family GPIO Interrupt Multiplexer - Double Edge Hack
+
+endif
diff --git a/drivers/amlogic/Makefile b/drivers/amlogic/Makefile
new file mode 100644
index 000000000000..1002a562e599
--- /dev/null
+++ b/drivers/amlogic/Makefile
@@ -0,0 +1,21 @@
+##########################################
+########## Amlogic Drivers ###############
+##########################################
+
+##########################################
+## The order of directories matter.
+## Do not change.
+##########################################
+
+ifndef CONFIG_KASAN
+KBUILD_CFLAGS += -Wlarger-than=28792
+KBUILD_CFLAGS += -Wstack-usage=1856
+else
+ifeq ($(call cc-ifversion, -lt, 0500, y), y)
+    $(error -----GCC VERSION TOO SMALL FOR KASAN -----)
+endif
+endif
+
+obj-$(CONFIG_AMLOGIC_TOUCHSCREEN_TLSC6X) += tlsc6x/
+
+obj-$(CONFIG_AMLOGIC_GPIO_IRQ) += irqchip/
